### PR TITLE
uid exceptions fix

### PR DESCRIPTION
**Pull request by AndreiBozantan**
The code was modified to avoid the failing query for `user_id` columns. I also removed the `IsolatorCheckComponent` which became unnecessary.

### CODE REVIEW

File: tests/explorer.tests/ComponentTests.cs [hunk 1]
```diff
@@ -26,6 +27,7 @@ public async Task TestMinMaxRefinerComponent()
                 "gda_banking",
                 "loans",
                 "amount",
+                new ColumnInfo(DValueType.Integer, ColumnInfo.ColumnType.Regular),
```
Comment (dandanlen) [inline tests/explorer.tests/ComponentTests.cs:30]: We should probably query for this (maybe in the ExplorerTestFixture, that we we can cache the data for all tests). Hard-coding the values is a bit error-prone. 
This is not a big deal though, we can log an issue and do it at a later date.

File: src/explorer/Components/DistinctValuesComponent.cs [hunk 2]
```diff
@@ -78,9 +78,11 @@ public async IAsyncEnumerable<ExploreMetric> YieldMetrics()
 
         protected override async Task<Result> Explore()
         {
-            var distinctValueQ = await conn.Exec(
-                new DistinctColumnValues(ctx.Table, ctx.Column));
-
+            if (ctx.ColumnInfo.UserId)
+            {
+                return new Result(Enumerable.Empty<ValueWithCount<JsonElement>>());
```
[the remark this answers is not in the document]
Comment (AndreiBozantan) [reply src/explorer/Components/DistinctValuesComponent.cs:83]: I suppose that this metric will be correct in this context and more information will be actually helpful, so I am not sure that it makes sense to try removing it.

File: src/explorer/Common/ExplorerContext.cs [hunk 3]
```diff
@@ -10,6 +10,6 @@ public interface ExplorerContext
 
         public string Column { get; }
 
-        public DValueType ColumnType { get; }
+        public ColumnInfo ColumnInfo { get; }
```
[the remark this answers is not in the document]
Comment (AndreiBozantan) [reply src/explorer/Common/ExplorerContext.cs:13]: I wanted to keep all the column metadata grouped inside the `ColumnInfo` object. (The `Isolating` and `UserId` names are more meaningful when grouped inside the `ColumnInfo`, so I've put the column type inside that also).

File: src/explorer/Common/ColumnInfo.cs [hunk 4]
```diff
@@ -0,0 +1,45 @@
+namespace Explorer.Common
+{
+    using Diffix;
+
+    public class ColumnInfo
```
[the remark this answers is not in the document]
Comment (AndreiBozantan) [reply src/explorer/Common/ColumnInfo.cs:5]: Maybe, I also thought about it a little.

File: tests/explorer.tests/NumericDistributionTests.cs [hunk 5]
```diff
@@ -21,11 +23,12 @@ public async void TestNumericSampleGenerator()
                 "gda_banking",
                 "loans",
                 "duration",
+                new ColumnInfo(DValueType.Integer, ColumnInfo.ColumnType.Regular),
                 ExplorerTestFixture.GenerateVcrFilename(this));
 
             await scope.MetricsTest<NumericSampleGenerator>(result =>
             {
-                // TODO: Check metrics against schema definition
+                // TODO: Check metrics aga
```
[the remark this answers is not in the document]
Comment (AndreiBozantan) [reply tests/explorer.tests/NumericDistributionTests.cs:31]: seems like I was in a rush

File: src/explorer/Common/ColumnInfo.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (AndreiBozantan) [reply src/explorer/Common/ColumnInfo.cs]: Fixed with 60239a0.

File: tests/explorer.tests/NumericDistributionTests.cs (hunk 5 above)
[the remark this answers is not in the document]
Comment (AndreiBozantan) [reply tests/explorer.tests/NumericDistributionTests.cs]: Fixed with b186f0c.

File: tests/explorer.tests/ComponentTests.cs (hunk 1 above)
Comment (AndreiBozantan) [reply tests/explorer.tests/ComponentTests.cs]: I created the new issue #214 for this.

File: src/explorer/Components/DistinctValuesComponent.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (dandanlen) [reply src/explorer/Components/DistinctValuesComponent.cs:83]: But it's not correct - an id column is not categorical. Or am I missing something?

[the remark this answers is not in the document]
Comment (AndreiBozantan) [reply src/explorer/Components/DistinctValuesComponent.cs:83]: From my tests, I see that the metric published is:
```                
                {
                    "name": "distinct.is_categorical",
                    "value": false
                },
```
The value is **false**, which is correct. 
(Sorry, for the confusion, I didn't pay complete attention to your initial comment.)

[the remark this answers is not in the document]
Comment (dandanlen) [reply src/explorer/Components/DistinctValuesComponent.cs:83]: Ah, I got `true`...

The check is this:
```
            if (valueCounts.SuppressedRowRatio < SuppressedRatioThreshold)
            {
                [...]

                yield return new UntypedMetric(name: "distinct.is_categorical", metric: true);

                [...]
            }
            else
            {
                yield return new UntypedMetric(name: "distinct.is_categorical", metric: false);
            }
```

And if the results list is empty, then `valueCounts.SuppressedRowRatio` is `0` so the condition will always evaluate to `true`, no?

[the remark this answers is not in the document]
Comment (AndreiBozantan) [reply src/explorer/Components/DistinctValuesComponent.cs:83]: I checked the code more carefully and the `SuppressedRowRatio` is actually `NaN` due to division by zero:
`valueCounts.SuppressedRowRatio = (double)SuppressedCount / TotalRows;`,  where `TotalRows` is zero.

[the remark this answers is not in the document]
Comment (dandanlen) [reply src/explorer/Components/DistinctValuesComponent.cs:83]: You're right. 

Ok, I think I misread the results when I ran the test locally.

**Pull request merged**